### PR TITLE
Kishor issue 14663

### PR DESCRIPTION
**Pull request by KishorkumarKar**
### Description (*)

While updating customer through api customer group is changed to 1 if there no customer group in the body

### Fixed Issues (magento/magento2#14663)

1. magento/magento2#14663: Updating Customer through rest/all/V1/customers/:id resets group_id if group_id not passed in payload

### Manual testing scenarios (*)

1. Update any customer who's group_id isn't the first group_id in the source list
2. Pass the customer payload with email, firstname, lastname, website_id, id

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/Customer/Model/Customer.php [hunk 1]
```diff
@@ -883,6 +883,12 @@ public function sendPasswordResetConfirmationEmail()
      */
     public function getGroupId()
     {
+		if ($this->hasData('id') && !$this->hasData('group_id')) {
+			 $groupId=$customerDataObject= $this->_getResource()->getGroupId($this->getData('id'));			 
+			 if($groupId) {
+				 $this->setData('group_id', $groupId);
```
Comment (orlangur) [inline app/code/Magento/Customer/Model/Customer.php:889]: Getter must not change object state.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=22700) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=22700) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: KishorkumarKar<br/>:x: kishorInsync

File: app/code/Magento/Customer/Model/Customer.php (hunk 1 above)
Comment (KishorkumarKar) [reply app/code/Magento/Customer/Model/Customer.php:889]: I have remove $customerDataObject= is it ok

**Comment by sidolov** [comment]
Hi @KishorkumarKar  , please, sign CLA, otherwise, we can't process your pull request

**Comment by sidolov** [comment]
@KishorkumarKar , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**